### PR TITLE
chore(suite-native): button has ReactElement not only icon

### PR DESCRIPTION
**Pull request by vytick**
Make Button to accept not only IconName but also a ReactElement for both sides.

This will let us insert animated elements and I will be able to use that in Device switcher [Change button animation](https://www.figma.com/proto/mmDnYtctfjnreWhTjLfeQH/Passphrase?page-id=0%3A1&type=design&node-id=2146-55968&viewport=5976%2C4395%2C1.31&t=8i3EiJVzLmaibGWX-1&scaling=min-zoom&starting-point-node-id=2146%3A55968&show-proto-sidebar=1&mode=design&fuid=1293095861312443248).

Example of custom and also animated element:

https://github.com/trezor/trezor-suite/assets/2011829/1253af02-a3d9-4e48-920a-e9cb96cac62e

### CODE REVIEW

File: suite-native/atoms/src/Button/Button.tsx [hunk 1]
```diff
@@ -35,7 +40,7 @@ export type ButtonProps = Omit<PressableProps, 'style' | 'onPressIn' | 'onPressO
     size?: ButtonSize;
     style?: NativeStyleObject;
     isDisabled?: boolean;
-} & MergeExclusive<{ iconLeft?: IconName }, { iconRight?: IconName }> &
+} & MergeExclusive<{ viewLeft?: ButtonAccessoryType }, { viewRight?: ButtonAccessoryType }> &
```
Comment (PeKne) [inline suite-native/atoms/src/Button/Button.tsx]: Why did you change the name fro icon to view? the icon seems to be more explanatory to me. We still want to send there only icons (even though they can be animated now). 

But maybe this is just my personal preference.

File: suite-native/atoms/src/Button/Button.tsx [hunk 2]
```diff
@@ -1,18 +1,23 @@
...
+import { Icon, IconColor, IconName, IconSize, icons } from '@suite-common/icons';
 
 import { Text } from '../Text';
 import { useButtonPressAnimatedStyle } from './useButtonPressAnimatedStyle';
 import { TestProps } from '../types';
 import { HStack } from '../Stack';
 
+// Using ReactElement instead of ReactNode to exclude string and have type check on IconName
+// and also because string needs to be rendered in the <Text> element anyway
+export type ButtonAccessoryType = IconName | ReactElement | undefined | null;
```
Comment (PeKne) [inline suite-native/atoms/src/Button/Button.tsx]: Why is `undefined` and `null` needed here?

Comment (PeKne) [reply suite-native/atoms/src/Button/Button.tsx]: I think that type suffix is redundant here.
```suggestion
export type ButtonAccessory = IconName | ReactElement | undefined | null;
```

File: suite-native/atoms/src/Button/Button.tsx [hunk 3]
```diff
@@ -44,6 +49,12 @@ type ButtonIconProps = {
     buttonSize: ButtonSize;
 };
 
+type ButtonAccessoryViewType = {
```
Comment (PeKne) [inline suite-native/atoms/src/Button/Button.tsx]:
```suggestion
type ButtonAccessoryView = {
```

File: suite-native/atoms/src/Button/Button.tsx [hunk 4]
```diff
@@ -278,11 +307,15 @@ export const Button = ({
                 ]}
             >
                 <HStack alignItems="center">
-                    {iconLeft && icon}
+                    <ButtonAccessoryView element={viewLeft} iconColor={iconColor} iconSize={size} />
```
Comment (PeKne) [inline suite-native/atoms/src/Button/Button.tsx]: Why was the conditional rendering removed? IMO it is more readable like that.

File: suite-native/atoms/src/Button/Button.tsx [hunk 5]
```diff
@@ -227,9 +238,32 @@ export const ButtonIcon = ({ iconName, color, buttonSize }: ButtonIconProps) =>
...
+export const ButtonAccessoryView = ({
+    element,
+    iconColor = 'iconDefault',
+    iconSize = 'medium',
+}: ButtonAccessoryViewType) => {
+    if (isIconName(element)) {
+        if (!iconColor || !iconSize) return null;
+
+        return (
+            <ButtonIcon iconName={element as IconName} color={iconColor} buttonSize={iconSize} />
```
Comment (PeKne) [inline suite-native/atoms/src/Button/Button.tsx]: Change the `isIconName` to a type guard (`const isIconName = (value: ButtonAccessoryType) : is IconName => ...`) more here: https://www.typescriptlang.org/docs/handbook/advanced-types.html#user-defined-type-guards. 

Then you can remove the type assertion here.

```suggestion
            <ButtonIcon iconName={element} color={iconColor} buttonSize={iconSize} />
```

File: suite-native/atoms/src/Button/Button.tsx [hunk 6]
```diff
@@ -227,9 +238,32 @@ export const ButtonIcon = ({ iconName, color, buttonSize }: ButtonIconProps) =>
...
+
+// ButtonAccessoryView renders either a ButtonIcon or a provided custom element
+// iconColor and iconSize are only used when element is an IconName
+export const ButtonAccessoryView = ({
+    element,
+    iconColor = 'iconDefault',
+    iconSize = 'medium',
+}: ButtonAccessoryViewType) => {
+    if (isIconName(element)) {
+        if (!iconColor || !iconSize) return null;
```
Comment (PeKne) [inline suite-native/atoms/src/Button/Button.tsx]: This condition will be never used since the iconColor and IconSize have default values.

File: suite-native/atoms/src/Button/Button.tsx [hunk 7]
```diff
@@ -227,9 +238,32 @@ export const ButtonIcon = ({ iconName, color, buttonSize }: ButtonIconProps) =>
...
+export const isIconName = (element: ButtonAccessoryType) =>
+    G.isNotNullable(element) &&
+    typeof element === 'string' &&
+    icons[element as IconName] !== undefined;
+
+// ButtonAccessoryView renders either a ButtonIcon or a provided custom element
+// iconColor and iconSize are only used when element is an IconName
+export const ButtonAccessoryView = ({
+    element,
+    iconColor = 'iconDefault',
```
Comment (PeKne) [inline suite-native/atoms/src/Button/Button.tsx]: What about "hiding" these defaults inside of IconButton component? It should be its responsible IMO. Just idea.

**Review by PeKne — CHANGES_REQUESTED** [review]

File: suite-native/atoms/src/Button/Button.tsx (hunk 1 above)
Comment (vytick) [reply suite-native/atoms/src/Button/Button.tsx]: I changed the name because we can put anything there now, not only icon

File: suite-native/atoms/src/Button/Button.tsx (hunk 2 above)
Comment (vytick) [reply suite-native/atoms/src/Button/Button.tsx]: I put it there to handel undefined | null inside. But i will change it to conditional rendering as you suggested for the readibility

Comment (vytick) [reply suite-native/atoms/src/Button/Button.tsx]: 0c52c10957

File: suite-native/atoms/src/Button/Button.tsx (hunk 3 above)
Comment (vytick) [reply suite-native/atoms/src/Button/Button.tsx]: 0c52c10957

File: suite-native/atoms/src/Button/Button.tsx (hunk 4 above)
Comment (vytick) [reply suite-native/atoms/src/Button/Button.tsx]: 0c52c10957

File: suite-native/atoms/src/Button/Button.tsx (hunk 5 above)
Comment (vytick) [reply suite-native/atoms/src/Button/Button.tsx]: 0c52c10957

File: suite-native/atoms/src/Button/Button.tsx (hunk 6 above)
Comment (vytick) [reply suite-native/atoms/src/Button/Button.tsx]: 0c52c10957

File: suite-native/atoms/src/Button/Button.tsx (hunk 7 above)
Comment (vytick) [reply suite-native/atoms/src/Button/Button.tsx]: 0c52c10957

File: suite-native/atoms/src/Button/Button.tsx (hunk 2 above)
Comment (PeKne) [reply suite-native/atoms/src/Button/Button.tsx]: the `Type` suffix is still there.

Comment (vytick) [reply suite-native/atoms/src/Button/Button.tsx]: good point. updated in 705cf82150

**Review by PeKne — APPROVED** [review]
Thanks for addressing my request 🙌
